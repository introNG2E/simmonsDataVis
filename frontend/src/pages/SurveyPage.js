
import React, {Component, useState} from 'react';

import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import axios from 'axios';

import * as Survey from 'survey-react';
import data from '../data/survey.js';

import "survey-react/survey.css";
import '../styles/SurveyPage.css';
import {Button, Grid, Icon, Modal, Divider} from "semantic-ui-react";

class SurveyPage extends Component {
    // redirecting handles redirecting to data page
    constructor(props){
        super(props);
        this.state={
            redirecting: false
        }
    }
    json = data;
    constructor(props) {
        super(props);
        this.state = {
            modalIsOpen: false
        };
        this.openModal = this.openModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
        this.onHomePress = this.onHomePress.bind(this);
    }


    // Called when a question in the survey is changed/answered
    onValueChanged = (result) => {

        console.log("value change!");
    }

    // Called when survey is submitted
    onComplete = (result) => {
        // console.log("Complete! "+ JSON.stringify(result.data));

        // user is appended to route to make put request
        const id = this.props.id;
        console.log(this.props.id);
        axios.put(
            'http://localhost:8080/api/students/' + id,
            { survey: result.data },
            { headers: { 'Content-Type': 'application/json' } }
        )
        .then(res => {
            console.log(res);

            // If put request is successful, sets redirecting to true
            this.setState({
                redirecting: true
            })
        })
        .catch(err => {
            console.log(err);
        })

    };

    onHomePress(props) {
        this.props.history.push('/Homepage');
    }

    openModal() {
        console.log('wow');
        this.setState({modalIsOpen: true});
    }

    closeModal() {
        this.setState({modalIsOpen: false});
    }

    render() {
        const model = new Survey.Model(this.json);

        // Redirects to data page when survey is complete
        if(this.state.redirecting) {
            return <Redirect to='/data' />
        }

        return (
            <div className="surveyjs">
                <Survey.Survey
                    model={model}
                    onComplete={this.onComplete}
                    onValueChanged={this.onValueChanged}
                    className="survey"
                />
                <Divider fitted hidden/>
                <Grid>
                    <Grid.Row centered stretched>
                        <Modal open={this.state.modalIsOpen} size="tiny" trigger={<Button basic color = "black"centered = {true} onClick={this.openModal}>
                                <Icon name='home'/> Home
                            </Button>}
                    >
                        <Modal.Header>
                            <Grid>
                                <Grid.Row centered>
                                    If you return to the homepage, you will lose your progress. Do you wish to
                                    proceed?
                                </Grid.Row>
                            </Grid>
                        </Modal.Header>
                        <Modal.Content>
                            <Modal.Actions>
                                <Grid columns={2}>
                                    <Grid.Row centered>
                                        <Grid.Column>
                                            <Button  size = 'tiny' color='green' onClick={this.onHomePress}>
                                                <Icon name='checkmark'/> Yes
                                            </Button>
                                        </Grid.Column>
                                        <Button size = 'tiny' color='red' onClick={this.closeModal}>
                                            <Icon name='remove'/> No
                                        </Button>
                                    </Grid.Row>
                                </Grid>
                            </Modal.Actions>
                        </Modal.Content>
                    </Modal>
                    </Grid.Row>
                </Grid>
                <Divider hidden/>
                </div>
        );
    }
}

export default SurveyPage;