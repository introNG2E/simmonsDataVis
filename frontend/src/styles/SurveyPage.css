.survey {
  height: 15px;
  display: flex;
}

Survey {
  height: 85vh;
}