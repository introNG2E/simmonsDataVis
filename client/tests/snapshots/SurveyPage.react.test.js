import React from 'react';
import SurveyPage from '../../src/pages/SurveyPage';
import renderer from 'react-test-renderer';

it('renders correctly with id passed to it', async () => {
  const tree = renderer
    .create(<SurveyPage getId={'5de56e20ed26b55db0ba44c4'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without id passed to it', async () => {
  const tree = renderer
    .create(<SurveyPage getId={''} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});