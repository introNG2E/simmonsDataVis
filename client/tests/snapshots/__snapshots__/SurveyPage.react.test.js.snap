// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with id passed to it 1`] = `
<div
  className="surveyjs"
>
  <div
    className="sv_main sv_default_css"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="sv_custom_header"
      />
      <div
        className="sv_container"
      >
        <div
          className="sv_header"
        >
          <h3>
            <span
              style={null}
            >
              Engineering Out-Of-Class Activity Survey
            </span>
          </h3>
          <h5>
            <span
              style={null}
            >
              
            </span>
          </h5>
        </div>
        <div
          className="sv_body"
          id="sp_100"
        >
          <div
            className="sv_progress"
            style={Object {}}
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              className="sv_progress_bar"
              role="progressbar"
              style={
                Object {
                  "width": "17%",
                }
              }
            >
              <span
                className="undefined sv-hidden"
              >
                Page 1 of 6
              </span>
            </div>
            <span
              className="undefined undefined"
            >
              Page 1 of 6
            </span>
          </div>
          <div
            className="sv_p_root"
          >
            <h4
              className="sv_page_title"
            >
              <span
                style={null}
              >
                
              </span>
            </h4>
            <div
              className=""
            >
              <span
                style={null}
              >
                
              </span>
            </div>
            <div
              className="sv_row"
            >
              <div
                className="sv_p_container"
                style={
                  Object {
                    "width": "100.000000%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "display": "block",
                      "paddingLeft": "",
                    }
                  }
                >
                  <div
                    className="sv_row"
                  >
                    <div
                      className="sv_q sv_qstn"
                      id="sq_100"
                      style={
                        Object {
                          "width": "100.000000%",
                        }
                      }
                    >
                      <div
                        className=""
                      >
                        <div
                          className=""
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<p>Study Title: Investigating Co-Curricular Participation of Students Underrepresented in Engineering</p>


<p>Researchers: Anh Chau</p>
<h4><u>Consent Information and Directions</u></h4><p>Do you participate in any activities? If so, I would like to know more about these activities. You are invited to participate in a research study that seeks to learn about students’ experiences from participating in various activities, called co/extracurricular activities, which occur outside of taking courses.</p><h4><u>Consent Information</u></h4><p>The information you provide will be used to help college and university administrators and faculty improve out of class experiences and students' ability to benefit from such experiences. It may also identify links between participation in out of class activities and commitment to engineering majors and careers.</p><p>It should take about 20 minutes to complete the survey. Persons who complete the survey will be entered into a drawing for ten (10) $25 Amazon gift cards. The odds of winning are 1 in 200.  To enter the drawing, please add your name and contact information at the end of the survey.  You are not required to enter the drawing.</p><p>There are no right or wrong answers. However, please choose your response options carefully. There are no known risks to study participants. Your responses will remain confidential. You are free to discontinue this survey at any time without  any consequences. You should be over 18 to be eligible to participate in this study. Completion of this online questionnaire will be considered as your consent to participate in this study and permission to use your data for the purposes of the study specified above.</p><p>Any questions concerning this study should be directed to Dr. Denise R. Simmons at densimm@vt.edu or (540) 553-6013. The Institutional Review Board (IRB) at Virginia Tech (VT) has approved this study. Should you have any questions or concerns about the study’s conduct or your rights as a research subject, or need to report a research-related injury or event, you may contact the VT IRB Chair, Dr. David M. Moore atmoored@vt.edu or (540) 231-4991.</p><h5><u>Directions:</u></h5><p>(1) You can use the back button to move Back and forth between pages.</p><p>(2) Hovering over words in <font color=\\"blue\\">blue ink</font> will display text, specifically examples of out-of-classroom activities and outcomes.</p><p><strong>(3) Clicking \\"agree\\" below indicates that:
</strong>
</p>
<p><b>- you voluntarily agree to participate</b></p>
<p><b>- you are at least 18 years of age</b></p>

<p><b>If you do not wish to participate in the study, please decline by clicking the \\"disagree\\" button. </b></p><p></p>",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="sv_row"
                  >
                    <div
                      className="sv_q sv_qstn"
                      id="sq_101"
                      style={
                        Object {
                          "width": "100.000000%",
                        }
                      }
                    >
                      <div
                        className=""
                      >
                        <div
                          className=""
                        >
                          <div
                            className="sv_select_wrapper"
                          >
                            <select
                              aria-label="consentQuestion"
                              className="sv_q_dropdown_control"
                              id="sq_101i"
                              onChange={[Function]}
                              onInput={[Function]}
                              value=""
                            >
                              <option
                                value=""
                              >
                                Choose...
                              </option>
                              <option
                                disabled={false}
                                value="agree"
                              >
                                Agree
                              </option>
                              <option
                                disabled={false}
                                value="disagree"
                              >
                                Disagree
                              </option>
                            </select>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="sv_nav"
          >
            <input
              className=" sv_next_btn"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "marginRight": "5px",
                }
              }
              type="button"
              value="Next"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    className="ui hidden divider"
  />
  <div
    className="ui grid"
  >
    <div
      className="centered stretched row"
    >
      <button
        centered="true"
        className="ui black basic button"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="home icon"
          onClick={[Function]}
        />
         Home
      </button>
    </div>
  </div>
  <div
    className="ui hidden divider"
  />
</div>
`;

exports[`renders correctly without id passed to it 1`] = `
<div
  className="surveyjs"
>
  <div
    className="sv_main sv_default_css"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="sv_custom_header"
      />
      <div
        className="sv_container"
      >
        <div
          className="sv_header"
        >
          <h3>
            <span
              style={null}
            >
              Engineering Out-Of-Class Activity Survey
            </span>
          </h3>
          <h5>
            <span
              style={null}
            >
              
            </span>
          </h5>
        </div>
        <div
          className="sv_body"
          id="sp_120"
        >
          <div
            className="sv_progress"
            style={Object {}}
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              className="sv_progress_bar"
              role="progressbar"
              style={
                Object {
                  "width": "17%",
                }
              }
            >
              <span
                className="undefined sv-hidden"
              >
                Page 1 of 6
              </span>
            </div>
            <span
              className="undefined undefined"
            >
              Page 1 of 6
            </span>
          </div>
          <div
            className="sv_p_root"
          >
            <h4
              className="sv_page_title"
            >
              <span
                style={null}
              >
                
              </span>
            </h4>
            <div
              className=""
            >
              <span
                style={null}
              >
                
              </span>
            </div>
            <div
              className="sv_row"
            >
              <div
                className="sv_p_container"
                style={
                  Object {
                    "width": "100.000000%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "display": "block",
                      "paddingLeft": "",
                    }
                  }
                >
                  <div
                    className="sv_row"
                  >
                    <div
                      className="sv_q sv_qstn"
                      id="sq_151"
                      style={
                        Object {
                          "width": "100.000000%",
                        }
                      }
                    >
                      <div
                        className=""
                      >
                        <div
                          className=""
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<p>Study Title: Investigating Co-Curricular Participation of Students Underrepresented in Engineering</p>


<p>Researchers: Anh Chau</p>
<h4><u>Consent Information and Directions</u></h4><p>Do you participate in any activities? If so, I would like to know more about these activities. You are invited to participate in a research study that seeks to learn about students’ experiences from participating in various activities, called co/extracurricular activities, which occur outside of taking courses.</p><h4><u>Consent Information</u></h4><p>The information you provide will be used to help college and university administrators and faculty improve out of class experiences and students' ability to benefit from such experiences. It may also identify links between participation in out of class activities and commitment to engineering majors and careers.</p><p>It should take about 20 minutes to complete the survey. Persons who complete the survey will be entered into a drawing for ten (10) $25 Amazon gift cards. The odds of winning are 1 in 200.  To enter the drawing, please add your name and contact information at the end of the survey.  You are not required to enter the drawing.</p><p>There are no right or wrong answers. However, please choose your response options carefully. There are no known risks to study participants. Your responses will remain confidential. You are free to discontinue this survey at any time without  any consequences. You should be over 18 to be eligible to participate in this study. Completion of this online questionnaire will be considered as your consent to participate in this study and permission to use your data for the purposes of the study specified above.</p><p>Any questions concerning this study should be directed to Dr. Denise R. Simmons at densimm@vt.edu or (540) 553-6013. The Institutional Review Board (IRB) at Virginia Tech (VT) has approved this study. Should you have any questions or concerns about the study’s conduct or your rights as a research subject, or need to report a research-related injury or event, you may contact the VT IRB Chair, Dr. David M. Moore atmoored@vt.edu or (540) 231-4991.</p><h5><u>Directions:</u></h5><p>(1) You can use the back button to move Back and forth between pages.</p><p>(2) Hovering over words in <font color=\\"blue\\">blue ink</font> will display text, specifically examples of out-of-classroom activities and outcomes.</p><p><strong>(3) Clicking \\"agree\\" below indicates that:
</strong>
</p>
<p><b>- you voluntarily agree to participate</b></p>
<p><b>- you are at least 18 years of age</b></p>

<p><b>If you do not wish to participate in the study, please decline by clicking the \\"disagree\\" button. </b></p><p></p>",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="sv_row"
                  >
                    <div
                      className="sv_q sv_qstn"
                      id="sq_152"
                      style={
                        Object {
                          "width": "100.000000%",
                        }
                      }
                    >
                      <div
                        className=""
                      >
                        <div
                          className=""
                        >
                          <div
                            className="sv_select_wrapper"
                          >
                            <select
                              aria-label="consentQuestion"
                              className="sv_q_dropdown_control"
                              id="sq_152i"
                              onChange={[Function]}
                              onInput={[Function]}
                              value=""
                            >
                              <option
                                value=""
                              >
                                Choose...
                              </option>
                              <option
                                disabled={false}
                                value="agree"
                              >
                                Agree
                              </option>
                              <option
                                disabled={false}
                                value="disagree"
                              >
                                Disagree
                              </option>
                            </select>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="sv_nav"
          >
            <input
              className=" sv_next_btn"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "marginRight": "5px",
                }
              }
              type="button"
              value="Next"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    className="ui hidden divider"
  />
  <div
    className="ui grid"
  >
    <div
      className="centered stretched row"
    >
      <button
        centered="true"
        className="ui black basic button"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="home icon"
          onClick={[Function]}
        />
         Home
      </button>
    </div>
  </div>
  <div
    className="ui hidden divider"
  />
</div>
`;
